doc/source/f2py/usage.rst

```diff
@@ -195,7 +207,7 @@ Execute ``f2py`` without any options to get an up-to-date list of
 available options.
 
 Python module ``f2py2e``
-=========================
+========================
```

Comment:
What to do with the section describing `f2py2e`? Can we deprecate it altogether, or at least state why one would use this over `f2py`

Reply:
I think the  `f2py2e` module needs its own module documentation. Everything else calls through `f2py2e`, so it cannot be removed or deprecated, but as it is not directly called I don't think it serves any purpose to document it here. Note that the module needs updating to use current command line parsing and to document the functions it contains, but that is a task outside the scope of this PR.

Reply:
Looking at the `__init__.py` file, I wonder if we could change
```
c = '{} -c "import numpy.f2py as f2py2e;f2py2e.main()" {}'
```
to
```
c = '{} -m numpy.f2py  {}'
```

Reply:
Changing `f2py2e` -> `numpy.f2py`, seems reasonable, doctests work. We need #11259 to make the `compile` command work.